### ISSUE TITLE
Document source messed up when editing a page after an edit to a previous page

### ISSUE THREAD

**Issue by ktraunmueller** [ktraunmueller/Compositor#530]
Something's going awry when creating a document using the "Standard Report" template.

**Steps to reproduce** (Compositor 1.14):
- Create a new document using the "Standard Report" template
- Double-click the default title ("Title") and change it to something else (e.g., "CV")
- Scroll to the page that has the "1.1 Section" header
- Double-click the section header (so that "Section" is highlighted) and type one character
- The document source is corrupted at this point, and the document fails to typeset:
```
\documentclass[12pt,a4paper]{report}

\usepackage[utf8]{inputenc}
\usepackage{geometry}
\usepackage{lmodern}
\usepackage[T1]{fontenc}
\usepackage{graphicx}
\graphicspath{{./}}
\usepackage{csquotes}

\begin{document}

\title{CV}

\author{
Author \\
Department \\
University
}

\maketitle

\tableofcontents

\chapter{Chapter}

\section{Sec}\end{

T}
```

**Comment by ktraunmueller** [comment]
So this was not a problem limited to the Report document class, but a general severe bug in the way Compositor updates page snapshots.

The problem would show up any time you'd do something like this:
- edit any page in a document
- scroll to some page further back in the document 
- edit this page

At this point, the snapshot for the page you are trying to edit is outdated, and any edits would mess up the document source.

**Comment by ktraunmueller** [comment]
Fixed in 1.17

**Comment by ktraunmueller** [comment]
Fixed in 1.17
